### PR TITLE
ENG-10994: update windowing example with all partition APIs

### PR DESCRIPTION
**Pull request by zeemanhe**

### CODE REVIEW

**Comment by benjaminballard** [comment]
This looks good, and much much simpler.  One question I have is why is the PartitionDataTracker still necessary?  It seems all it is doing now is keeping a count of how many partitions there are, but I don't see where that is used.

**Comment by zeemanhe** [comment]
The partition size is used to calculate the target rows per partition, which is then used in procedure DeleteOldestToTarget.  I agree that  PartitionDataTracker is not necessary here.  The target rows per partition can be initialized and and continuously updated with every delete. Thanks to the all partition call API which returns # of partitions.  PartitionDataTracker is removed.

File: examples/windowing/client/windowing/ContinuousDeleter.java [hunk 1]
```diff
@@ -55,118 +52,87 @@
...
-                long tuplesDeleted = clientResponse.getResults()[0].asScalarLong();
-                app.addToDeletedTuples(tuplesDeleted);
-                // If the procedue deleted up to its limit, reduce the time before
-                // the deletes process runs again.
-                if (tuplesDeleted >= app.config.deletechunksize) {
-                    unifinished.set(true);
-                }
+        public void clientCallback(ClientResponseWithPartitionKey[] clientResponse) throws Exception {
+
+            if (clientResponse == null) {
```
Comment (jhugg) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: Why should this ever be null?

File: examples/windowing/client/windowing/ContinuousDeleter.java [hunk 2]
```diff
@@ -55,118 +52,87 @@
...
-                // Un-comment to see problems on the console
-                //System.err.println(clientResponse.getStatusString());
 
-                // Track failures in a simplistic way.
-                failureCount.incrementAndGet();
+            app.updatePartitionInfo(clientResponse.length);
+
+            for (ClientResponseWithPartitionKey resp: clientResponse) {
+
+                if (resp.getResponse() != null && resp.getResponse().getStatus() == ClientResponse.SUCCESS) {
```
Comment (jhugg) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: Why would resp.getResponse() ever be null?

Comment (jhugg) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: For that matter, I thought we were getting rid of accessors? Did those get put back in?

File: examples/windowing/client/windowing/ContinuousDeleter.java [hunk 3]
```diff
@@ -55,118 +52,87 @@
...
-                }
-                catch (Exception e) {
-                    // Track failures in a simplistic way.
-                    failureCount.incrementAndGet();
-                    // Make sure the latch is released for this call, even if failed
-                    latch.countDown();
-                }
+            // Send the procedure call to all partitions and get results from each partitions.
+            if (app.config.historyseconds > 0) {
+                app.client.callAllPartitionProcedure(new CallBack(unfinished),
```
Comment (jhugg) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: Where do we actually wait for the callback to be called? We just check unfinished below, but it's unlikely this proc will be done. The old version uses a latch.

Why use the async api at all? Why not just use the sync API?

File: examples/windowing/client/windowing/WindowingApp.java [hunk 4]
```diff
@@ -162,9 +148,9 @@ TimestampType getTargetDate() {
         return new TimestampType(targetTimestampMillis * 1000);
     }
 
-    void updatePartitionInfo(Map<Long, PartitionInfo> partitionData) {
-        this.partitionData.set(partitionData);
-        targetRowsPerPartition.set(config.maxrows / partitionData.size());
+    void updatePartitionInfo(long partitionSize) {
```
Comment (jhugg) [inline examples/windowing/client/windowing/WindowingApp.java]: Maybe this should be called "updatePartitionCount".

File: examples/windowing/client/windowing/WindowingApp.java [hunk 5]
```diff
@@ -313,18 +297,19 @@ public void run() {
         System.out.println(" Starting Processing");
         System.out.println(HORIZONTAL_RULE);
 
+        try {
```
Comment (jhugg) [inline examples/windowing/client/windowing/WindowingApp.java]: So this code is just to get started? There should be a comment. Also, what if it fails? Should the program stop? Currently it just keeps going.

File: examples/windowing/client/windowing/ContinuousDeleter.java (hunk 2 above)
Comment (xinjiacs) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: What's the reason not to add accessor?

Comment (jhugg) [inline examples/windowing/client/windowing/ContinuousDeleter.java]: Less code. Make it a struct-y thing, not a fancy class.

**Comment by jhugg** [comment]
This looks a lot better to me. Thanks.

**Pull request merged**